### PR TITLE
Messenger: session and data table refactoring

### PR DESCRIPTION
**Pull request by SKuipers**
Refactored the Manage Messages data table and all $_SESSION variables.

### CODE REVIEW

File: modules/Messenger/messenger_manage.php [hunk 1]
```diff
@@ -19,461 +19,242 @@
...
+    $page->addError(__('You do not have access to this action.'));
+} else {
+    // Get action with highest precedence
+    $highestAction = getHighestGroupedAction($guid, $_GET["q"], $connection2);
+    if ($highestAction==FALSE) {
+        $page->addError(__('The highest grouped action cannot be determined.'));
+        return;
+    }
+
+    $search = isset($_GET['search'])? $_GET['search'] : '';
```
Comment (rossdotparker) [inline modules/Messenger/messenger_manage.php:38]: Could this not be a null coalesce?

File: src/Domain/Messenger/MessengerGateway.php [hunk 2]
```diff
@@ -35,8 +35,35 @@ class MessengerGateway extends QueryableGateway
 
     private static $tableName = 'gibbonMessenger';
     private static $primaryKey = 'gibbonMessengerID';
-    private static $searchableColumns = [''];
+    private static $searchableColumns = ['gibbonMessenger.subject', 'gibbonMessenger.body'];
     
+    /**
+     * Queries the list of users for the Manage Users page.
```
Comment (rossdotparker) [inline src/Domain/Messenger/MessengerGateway.php:41]: Looks like a reused doc block. Maybe worth noting that this does not determine which messages to display on, say, the message wall, but is for managing messages.

**Comment by rossdotparker** [comment]
Two refactorings in one : ) I'll merge this now, but there are two comments if you could pick them up post merge. Thanks @SKuipers!

**Comment by SKuipers** [comment]
Thanks! Well caught on those two spots 👍

**Pull request merged**